addons/binding/org.openhab.binding.keba/src/main/java/org/openhab/binding/keba/handler/KeContactP20Handler.java

```diff
@@ -0,0 +1,967 @@
+		response = StringUtils.chomp(response);
+
+		if (response.contains("TCH-OK")) {
+			// ignore confirmation messages which are not JSON
+			return;
+		}
+
+		JsonParser parser = new JsonParser();
+		try {
+			JsonObject readObject = parser.parse(response).getAsJsonObject();
+
+			for (Entry<String, JsonElement> entry : readObject.entrySet()) {
+
+				switch (entry.getKey()) {
+				case "Product": {
+					State newState = new StringType(entry.getValue()
+							.getAsString());
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_MODEL), newState);
+					break;
+				}
+				case "Serial": {
+					State newState = new StringType(entry.getValue()
+							.getAsString());
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_SERIAL), newState);
+					break;
+				}
+				case "Firmware": {
+
+					State newState = new StringType(entry.getValue()
+							.getAsString());
+					updateState(new ChannelUID(getThing().getUID(),
+							CHANNEL_FIRMWARE), newState);
+					break;
+
+				}
+				case "Plug": {
+					int state = entry.getValue().getAsInt();
+					switch (state) {
+					case 0: {
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_WALLBOX), OnOffType.OFF);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_VEHICLE), OnOffType.OFF);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_PLUG_LOCKED), OnOffType.OFF);
+						break;
+					}
+					case 1: {
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_WALLBOX), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_VEHICLE), OnOffType.OFF);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_PLUG_LOCKED), OnOffType.OFF);
+						break;
+					}
+					case 3: {
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_WALLBOX), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_VEHICLE), OnOffType.OFF);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_PLUG_LOCKED), OnOffType.ON);
+						break;
+					}
+					case 5: {
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_WALLBOX), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_VEHICLE), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_PLUG_LOCKED), OnOffType.OFF);
+						break;
+					}
+					case 7: {
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_WALLBOX), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_VEHICLE), OnOffType.ON);
+						updateState(new ChannelUID(getThing().getUID(),
+								CHANNEL_PLUG_LOCKED), OnOffType.ON);
+						break;
+					}
+					}
+					break;
+				}
+				case "State": {
+					int state = entry.getValue().getAsInt();
+					String result = "";
+					switch (state) {
+					case 0:
+						result = "Starting";
```

Comment:
actually, the channel state should be a Number and you should put the label texts like "Starting" into the options of the state description - this will allow internationalization.